### ISSUE TITLE
Javascript Error with the Project Generator

### ISSUE THREAD

**Issue by ofTheo** [openframeworks/projectGenerator#218]
see: https://github.com/openframeworks/openFrameworks/issues/6382 

Note: I don't see this issue with the nightly on macOS. 

This issue might be related with:
#205
#206 
#207

**Comment by ofTheo** [comment]
ping @hiroMTB 

If there is an easy fix to fix the error without losing the new IDE selection support would be great to do a PR so we can do a release.  

Thanks!

**Comment by rossanag** [comment]
I have to say after download the projectGenerator project from github and installed it, the message disappeared, however I am not able to pick a platform, by default XCode is set and I can't change it. Plus, I am using Ubuntu 19.04. Thanks!

**Comment by hiroMTB** [comment]
Hello, thanks for testing my update. 
As @bakercp mentioned in https://github.com/openframeworks/openFrameworks/issues/6382, `npm  install` or `npm update` should solve this issue.  (Or maybe we need to update /frontend/npm-shrinkwrap.json)

This is because of new npm nodule named 'n-readline'. I used this to read template.config files at [here](https://github.com/openframeworks/projectGenerator/blob/550d7467b6391361b30cd549ce5daf1b486c807f/frontend/index.js#L582) .

Right now I compiled frontend on macOS and Window 10 and I dont see any warning popup. The platform dropdown UI works ok too. Last time I tested on Ubuntu 16.04 but I can not test right now. 

Could you please run `npm update` and if it works we can add this line in build script.

**Comment by hiroMTB** [comment]
To be short,

- Solution A
Execute `npm update` and push changes on /frontend/npm-shrinkwrap.json

- Solution B
Add `npm update` to **create_package.sh**.
Around this lines 
https://github.com/openframeworks/openFrameworks/blob/aa2f917ed29e357721c02e1f10e87398081a4472/scripts/dev/create_package.sh#L391-L483

**Comment by hiroMTB** [comment]
I sent PR above.
I think Solution A is better. We can make PG frontend identical.
Solution B tries to use latest npm package, so each app will be different depends on the timing of generation.

**Comment by arturoc** [comment]
Thanks @hiroMTB, @rossanag can you check that your error is also gone after this change? should be in tomorrows nightly build

**Comment by hiroMTB** [comment]
Thanks for merge, I can check tmrw's nightly bit only for osx and win.

**Comment by rossanag** [comment]
Sorry for the late response, the error message is gone, the only issue I have now is that the MAC platform is set by default and I can choose another one.
The only way to get it working is using command line

>  projectGenerator -o"/path/toOF" -a"ofxGui ofxCv ofxNI2" -t"vscode" /home/rossig/openFrameworks/apps/myApps/NI2

**Comment by hiroMTB** [comment]
@rossanag
Thank you for checking.
It seems latest nightly is not ready yet, I will send you ping again.

About the default platform issue, I will take a look.

**Comment by hiroMTB** [comment]
@arturo 
Nightly is still old, does it fail? Where can we check build log?

**Comment by arturoc** [comment]
hey, yeah the problem is nightlies need a commit to the main repo, otherwise the server considers there's no change and it doesn't trigger the nightly. Yesterday there was a change but now the main repo is broken cause of the new glm version. Once those are fixed the nightlies will get built properly

**Comment by hiroMTB** [comment]
Ah thanks!
I thought it compiles every night via cron or something because of its name.

**Comment by rossanag** [comment]
> @rossanag
> Thank you for checking.
> It seems latest nightly is not ready yet, I will send you ping again.
> 
> Edit: Please check "advanced options" checkbox. Then you can select and deselect platforms.
> 
> About the default platform issue, I will take a look.

You are right, my bad, after checing "Advanced options" I could select another platform except the option for VSCode. 
I think getting it enabled by default would be more comfortable.

I can select VSCode platform from the command line. I think I followed all the steps for get it included into the frontend.

**Comment by hiroMTB** [comment]
Thanks!
I also noticed that default platform is set properly if we download PG from
nightly or official zip. OSX is set as default for all OS but only for
development environment (compile from source code).

**Comment by ofTheo** [comment]
@hiroMTB @rossanag maybe you could check the the nightly when it builds tonight? 
We're hoping to do a Release soon so would be good to make sure the PG is working with your changes. 

Note: nightlies were broken again but I think should be good tonight - needed node to be updated to match the shrinkwrap.js update. 

Thanks!
Theo

**Comment by rossanag** [comment]
@ofTheo I will check it briefly, maybe later on. Best!

**Comment by rossanag** [comment]
After a few days I could try the last nightbuilds. The projectGenerator is working fine in Linux in the sense that it generate the project according the selected template (vscode, qtCreator, etc). No pop up windows error. The only think I noticed after a project is generated, no matter what template was choosen is that when I select Open IDE, it actually opens the .qbs file but not the project itself within the IDE.  It' s a minor detail but worth to mention, anyway the project file can be open with the proper IDE either from the selected IDE or by double click on the project file.

**Comment by rossanag** [comment]
Hello I was trying PG, it works fine in the GUI version, in the sense it creates the project file according the template selected (qtCreator, vscode, etc). But after creating the project if I select the option to open the IDE, it opens the .qbs file no matter which template I select. Anyway the project file can be open inside the proper IDE or by double clicking the project file created into myApps folder, and then the IDE opens.
Despite I create a vscode project, a .qbs file project is created also.

The popup windows with errors have disappeared.

**Comment by hiroMTB** [comment]
Thanks!
I also tested under following platforms and can confirm js error disappeared.

- macOS Mojave
- Windows 10
- Ubuntu 16.04 LTS 64bit gcc5

@rossanag
Thank you for your catch. This is a bit difficult situation right now. For example, when the user selects Xcode and vscode at the same time, we can not decide which IDE to open. The fundamental reason for this is that we use template feature to support vscode. Originally I wanted to put vscode option in the platform dropdown list. However, we noticed vscode does not specify any platform because it supports all platform. Right now the platform dropdown is made of mix of "OS" and "IDE". Probably we need to separate it to make thing clear.
